cas-server-support-ldap/src/main/java/org/jasig/cas/adaptors/ldap/services/LdapServiceRegistryDao.java

```diff
@@ -100,23 +99,23 @@ public void init() {
 
     @Override
     public RegisteredService save(final RegisteredService rs) {
-        if (rs.getId() != RegisteredService.INITIAL_IDENTIFIER_VALUE) {
-            return update(rs);
-        }
 
-        Connection connection = null;
-        try {
-            connection = getConnection();
-            final AddOperation operation = new AddOperation(connection);
+        if (this.ldapServiceMapper != null && this.searchRequest != null) {

```

Comment:
Correct extra safety. Because features are bundled in the same module, we need checks like this. If we start breaking things down, then checks go away because you can expect that by mere inclusion of a module, its dependencies must be available and non null. 

Thus #1192